meta-balena-common/recipes-core/systemd/systemd-serialgetty.bbappend

```diff
@@ -1,18 +1,5 @@
-do_install_append() {
-    if ${@bb.utils.contains('DISTRO_FEATURES','development-image','false','true',d)}; then
-        if [ ! -z "${SERIAL_CONSOLES}" ] ; then
-            tmp="${SERIAL_CONSOLES}"
-            for entry in $tmp ; do
-                    baudrate=`echo $entry | sed 's/\;.*//'`
-                    ttydev=`echo $entry | sed -e 's/^[0-9]*\;//' -e 's/\;.*//'`
-                    if [ "$baudrate" = "$default_baudrate" ] ; then
-                            # disable the service
-                            rm -rf ${D}${sysconfdir}/systemd/system/getty.target.wants/serial-getty@$ttydev.service
-                    else
-                            # disable the service
-                            rm -rf ${D}${sysconfdir}/systemd/system/getty.target.wants/serial-getty$baudrate@$ttydev.service
-                    fi
-            done
-        fi
-    fi
+do_install_prepend() {
+    sed -i '/^\[Unit\]/a ConditionPathExists=/var/volatile/development-features' ${WORKDIR}/serial-getty@.service
+    sed -i '/^\[Unit\]/a After=development-features.service' ${WORKDIR}/serial-getty@.service
+    sed -i '/^\[Unit\]/a Requires=development-features.service' ${WORKDIR}/serial-getty@.service
```

Comment:
In the future this may be better as a drop-in unit file rather than injection via sed, but no need to hold up this PR for it.

Reply:
@klutchell do drop-ins work with unit templates?

Reply:
@alexgg yeah, something like the following should work:
```
/lib/systemd/system/serial-getty.service.d/development-features.conf
```
```
[Unit]
ConditionPathExists=/var/volatile/development-features
After=development-features.service
Requires=development-features.service
```

Reply:
thanks @klutchell I'll make the change on a separate PR